kafka-ui-react-app/src/components/common/Input/Input.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { useFormContext, RegisterOptions } from 'react-hook-form';
+
+import StyledIcon from './InputIcon.styled';
+import StyledInput from './Input.styled';
+
+export interface InputProps
+  extends React.InputHTMLAttributes<HTMLInputElement> {
```

Comment:
You can export Props of `StyledInput` as `StyledInputProps` and extend them here to avoid the redefinition of style props like `inputSize`

Reply:
`StyledInput` has props that `Input` does not need, like `hasLeftIcon`. We could omit this prop entirely and define the `leftIcon` prop in the `StyledInput` and extend `Input` props from it, but I believe that would make the code much less readable. `rightIcon` is defined in the `Input` and `leftIcon` in `StyledInput` - looks kinda confusing to me.

Reply:
In such cases you can use generic `Omit<StyledInputProps, 'hasLeftIcon'>`.
I understand your awareness of ambiguity in this particular case, but in some components, it would be very useful to have the ability to pass specific styled props like `as`